src/components/course/Course.jsx

```diff
@@ -35,15 +36,15 @@ export default function Course() {
   // the URL query parameters and then remove it to keep the URLs clean.
   const algoliaSearchParams = useExtractAndRemoveSearchParamsFromURL();
 
-  const [courseData, fetchError] = useAllCourseData({
+  const [courseData, courseRecommendations, fetchError] = useAllCourseData({
     courseKey,
     enterpriseConfig,
     courseRunKey,
   });
 
   const initialState = useMemo(
     () => {
-      if (!courseData) {
+      if (!courseData || !courseRecommendations) {
```

Comment:
would there be a case when courseRecommendations are empty object or null or maybe undefined because server sent no recommendations and the whole page wont be displayed?

Reply:
This check is here so the response from both APIs is returned, until then we show a loading spinner. The scenario you mentioned is unlikely to occur. The only time this would occur is when Discovery is down. Since we are already fetching a lot of info from Discovery to display on the course page, if the calls to Discovery fails then we won't be able to show the page anyway. For empty array, the page would still be displayed as `![]` returns `false`.